public/app/features/browse-dashboards/components/BrowseActions/BrowseActions.tsx

```diff
@@ -6,51 +6,99 @@ import { Button, useStyles2 } from '@grafana/ui';
 import appEvents from 'app/core/app_events';
 import { ShowModalReactEvent } from 'app/types/events';
 
+import {
+  useDeleteDashboardMutation,
+  useDeleteFolderMutation,
+  useMoveDashboardMutation,
+  useMoveFolderMutation,
+} from '../../api/browseDashboardsAPI';
 import { useActionSelectionState } from '../../state';
 
 import { DeleteModal } from './DeleteModal';
 import { MoveModal } from './MoveModal';
 
-export interface Props {}
+export interface Props {
+  // this is a complete hack to force a full rerender.
+  // TODO remove once we move everything to RTK query
+  onActionComplete?: () => void;
+}
 
-export function BrowseActions() {
+export function BrowseActions({ onActionComplete }: Props) {
   const styles = useStyles2(getStyles);
   const selectedItems = useActionSelectionState();
+  const [deleteDashboard] = useDeleteDashboardMutation();
+  const [deleteFolder] = useDeleteFolderMutation();
+  const [moveFolder] = useMoveFolderMutation();
+  const [moveDashboard] = useMoveDashboardMutation();
+  const selectedDashboards = Object.keys(selectedItems.dashboard).filter((uid) => selectedItems.dashboard[uid]);
+  const selectedFolders = Object.keys(selectedItems.folder).filter((uid) => selectedItems.folder[uid]);
+
+  const onDelete = async () => {
+    // Delete all the folders sequentially
+    // TODO error handling here
+    for (const folderUID of selectedFolders) {
+      await deleteFolder(folderUID).unwrap();
+    }
+
+    // Delete all the dashboards sequenetially
+    // TODO error handling here
+    for (const dashboardUID of selectedDashboards) {
+      await deleteDashboard(dashboardUID).unwrap();
+    }
+    onActionComplete?.();
+  };
+
+  const onMove = async (destinationUID: string) => {
+    // Move all the folders sequentially
+    // TODO error handling here
+    for (const folderUID of selectedFolders) {
+      await moveFolder({
+        folderUID,
+        destinationUID,
+      }).unwrap();
+    }
+
+    // Move all the dashboards sequentially
+    // TODO error handling here
+    for (const dashboardUID of selectedDashboards) {
+      await moveDashboard({
+        dashboardUID,
+        destinationUID,
+      }).unwrap();
+    }
+    onActionComplete?.();
+  };
 
-  const onMove = () => {
+  const showMoveModal = () => {
     appEvents.publish(
       new ShowModalReactEvent({
         component: MoveModal,
         props: {
           selectedItems,
```

Comment:
I'm not a fan of the modal events system, and rather we just rendered the modal components ourselves. But that's not an issue of _this_ PR so its fine.